### PR TITLE
added cpp md

### PR DESCRIPTION
**Pull request by sameer-manek**

### CODE REVIEW

**Review by ramizbalayil — APPROVED** [review]

**Review by hatim-developer — APPROVED** [review]

File: cpp.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+**function names -** use camelcase with first letter lower, eg:
+``` C++
+void doSomethingForMe ()
+{
+  // ..
+}
+```
+
+**function params -** optionally, you can precede params with p_ when params in function scope named similar to params to avoid confusion. params follow same naming convention as local variables.
```
Comment (rampr) [inline cpp.md]: no optionally please. Let's be decisive and go with that

File: cpp.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  // ..
+};
+```
+
+**local variable name -**  use snake type. eg
+``` C++
+std::string first_name = “sameer”
+```
+
+**local class variables -** should be camel case with first letter lower case, eg
```
Comment (rampr) [inline cpp.md]: camel case by definition means first letter is lower. the other one is upper camel case
https://en.wikipedia.org/wiki/Camel_case

File: cpp.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+**function params -** optionally, you can precede params with p_ when params in function scope named similar to params to avoid confusion. params follow same naming convention as local variables. 
+``` C++
+void doSomethingForMe (float p_img_height)
+{
+  img_height = static_cast<int>(p_img_height);
+}
+```
+
+**namespaces -** should be camel case:
```
Comment (rampr) [inline cpp.md]: upper camel case. By default camel case has the first letter in lower case

File: cpp.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+**Class name -** use camel case with first upper. eg
+``` C++
+class Person
+{
+  // ..
+};
+```
+
+**local variable name -**  use snake type. eg
```
Comment (rampr) [inline cpp.md]: Should be snake_case

File: cpp.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+**namespaces -** should be camel case:
+``` C++
+namespace CoreConfig 
+{
+    // ... 
+}
+```
+
+**constants -** identifiers should be all caps. eg:
```
Comment (rampr) [inline cpp.md]: with a _ separating words

File: cpp.md (hunk 1 above)
Comment (sameer-manek) [reply cpp.md]: I kept it so people can know such sol is there. while reading large functions which deal with lots of variables (in our case when vyshnav was trying to call func to main thread) it becomes hard to know where the variable is coming from and this saves some brain cells while reading code and tracking them down.

However, I'll remove this.

File: cpp.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+{
+  if (i % 2 == 0)
+  {
+    std:;cout << i << std::endl;
+  }
+}
+```
+however, you can still use inline branching, for eg:
+``` C++
+if (this->user.getAge() < Globals::MIN_AGE) return false;
```
Comment (rampr) [inline cpp.md]: is this okay without braces ? looks difficult to read

File: cpp.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+| Platforms                | iOS, android, amazon                                |
+| Build tools              | CMake (3.6+), gradle for android (5.x), pbx for iOS |
+| Compiler flags           | Warnings and Errors                                 |
+| Optional libs (external) | Boost                                               |
+
+We are also using CPP check to avoid any false positives in the project. You can easily install it on your system and learn more about it from - http://cppcheck.sourceforge.net/
+
+### Coding Style
+
+**Class name -** use camel case with first upper. eg
```
Comment (rampr) [inline cpp.md]: Please talk about braces style. Whether it should be in a newline or the same line etc..

Comment (ramizbalayil) [reply cpp.md]: It was decided in newline. Please correct me if I'm wrong @sameer-manek

File: cpp.md (hunk 6 above)
Comment (sameer-manek) [reply cpp.md]: as long as it is in same line, it is fine. many standards like google say the same, also I read a medium article somewhere which said the same thing. personally, i also find this fine to read.

File: cpp.md (hunk 7 above)
Comment (sameer-manek) [reply cpp.md]: I have, line 75

File: cpp.md (hunk 1 above)
Comment (rampr) [reply cpp.md]: Understand, but will be difficult for the pre-commit hook to check this.

File: cpp.md (hunk 6 above)
Comment (rampr) [reply cpp.md]: cool. Please share the medium article if you can

**Review by rampr — APPROVED** [review]

File: cpp.md (hunk 6 above)
Comment (sameer-manek) [reply cpp.md]: https://medium.com/@jonathanabrams/single-line-if-statements-2565c62ff492

**Review by nasimafeef — APPROVED** [review]

**Pull request merged**